File: backend/voting-app/src/main/java/com/tu/votingapp/entities/elections/CandidateEntity.java
package com.tu.votingapp.entities.elections;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "candidates")
@Data
@EqualsAndHashCode(callSuper = false)
@ToString()
@AllArgsConstructor
@NoArgsConstructor
public class CandidateEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "candidate_names")
    private String name;

    @Column(name = "bio")
    private String bio;

    @ManyToOne
    @JoinColumn(name = "election_id")
    private ElectionEntity election;

    @Column(name = "votes_count")
    private int votesCount;

    @Column(name = "image_uri")
    private String imageUri;

    @Column(name = "position")
    private String position;
}

File: frontend/android/app/src/main/java/com/example/android/entity/election/CandidateEntity.kt
package com.example.android.entity.election

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "candidates",
    foreignKeys = [
        ForeignKey(
            entity = ElectionEntity::class,
            parentColumns = ["id"],
            childColumns = ["election_id"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class CandidateEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @ColumnInfo(name = "name")
    val name: String,

    @ColumnInfo(name = "bio")
    val bio: String? = null,

    @ColumnInfo(name = "election_id")
    val electionId: Long,

    @ColumnInfo(name = "votes_count")
    val votesCount: Int = 0,

    @ColumnInfo(name = "image_uri")
    val imageUri: String? = null,

    @ColumnInfo(name = "position")
    val position: String? = null,

    @ColumnInfo(name = "party_id")
    val partyId: Long? = null

)